sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ece-request/_tests_/ece-process-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,185 @@
+import {
+  Application,
+  ApplicationStatus,
+  COEStatus,
+  DisbursementValue,
+  DisbursementValueType,
+  OfferingIntensity,
+} from "@sims/sims-db";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  saveFakeApplicationDisbursements,
+  saveFakeStudent,
+  createFakeDisbursementValue,
+  createFakeInstitutionLocation,
+} from "@sims/test-utils";
+import {
+  createTestingAppModule,
+  describeProcessorRootTest,
+  mockBullJob,
+} from "../../../../../../test/helpers";
+import { INestApplication } from "@nestjs/common";
+import { formatDate, QueueNames } from "@sims/utilities";
+import { DeepMocked } from "@golevelup/ts-jest";
+import { ECEProcessIntegrationScheduler } from "../ece-process-integration.scheduler";
+import { getUploadedFile } from "@sims/test-utils/mocks";
+import { RecordTypeCodes } from "@sims/integrations/institution-integration/ece-integration";
+import * as Client from "ssh2-sftp-client";
+import * as dayjs from "dayjs";
+
+describe(describeProcessorRootTest(QueueNames.ECEProcessIntegration), () => {
+  let app: INestApplication;
+  let processor: ECEProcessIntegrationScheduler;
+  let db: E2EDataSources;
+  let sftpClientMock: DeepMocked<Client>;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource, sshClientMock } =
+      await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    sftpClientMock = sshClientMock;
+    // Processor under test.
+    processor = app.get(ECEProcessIntegrationScheduler);
+  });
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+    // Ensures that every institution location on database has no integration.
+    // to allow eligible COE requestS to be generated with the data created
+    // for every specific scenario.
+    await db.institutionLocation.update(
+      { hasIntegration: true },
+      { hasIntegration: false },
+    );
+  });
+
+  it("Should process an ECE request file and when there are valid disbursements and applications.", async () => {
+    // Arrange
+    // Student with valid SIN.
+    const student = await saveFakeStudent(db.dataSource);
+    // Student application eligible for e-Cert.
+    const application = await saveFakeApplicationDisbursements(
+      db.dataSource,
+      {
+        student,
+        firstDisbursementValues: [
+          createFakeDisbursementValue(
+            DisbursementValueType.CanadaLoan,
+            "CSLP",
+            1122,
+          ),
+        ],
+        institutionLocation: createFakeInstitutionLocation(undefined, {
+          initialValue: { hasIntegration: true },
+        }),
+      },
+      {
+        offeringIntensity: OfferingIntensity.fullTime,
+        applicationStatus: ApplicationStatus.Enrolment,
+        firstDisbursementInitialValues: {
+          coeStatus: COEStatus.required,
+        },
+      },
+    );
+    application.studentNumber = "1234567789";
```

Comment:
Why do we need the `studentNumber`? That is nullable in `sims.applications`.

Reply:
Please change `buildECEFileDetail` to do not require `application.studentNumber` as it's not required for the e-Cert generation.

Reply:
Good point @andrepestana-aot. I also thought that student number was not required, until the E2E test failed due to the null value from `institutionStudentNumber` used by the following code in the method `getFixedFormat()` in [ece-file-detail.ts](https://github.com/bcgov/SIMS/blob/1ed1e86bf0b642c77f11609c7e411aee9d8f19b2/sources/packages/backend/libs/integrations/src/institution-integration/ece-integration/ece-files/ece-file-detail.ts#L52C14-L52C33):
```js
      record.appendWithEndFiller(
        this.institutionStudentNumber,
        12,
        SPACE_FILLER,
      );
```
In this case, `institutionStudentNumber` is `studentNumber` obtained from another method. When the value `institutionStudentNumber` is null,  the method [`appendWithEndFiller`](https://github.com/bcgov/SIMS/blob/1ed1e86bf0b642c77f11609c7e411aee9d8f19b2/sources/packages/backend/libs/utilities/src/string-builder.ts#L36-L47) will raise a type error as shown below. 
![image](https://github.com/user-attachments/assets/cb5f9a64-0732-451b-8490-38a1870b2e4c)

Setting a value for `studentNumber` for the application in the E2E test will make it pass. In short, student number should be a mandatory field for submitting student applications, and the ECE request process integration ensures that student number is required. If there is any further doubts about this, I think we can raise the question in DEV meetings and make corresponding changes in separate story.

Reply:
Ok, I see. That is required for the ECE integration. Thanks for checking. We can confirm that later with other DEVS.